scipy/optimize/_optimize.py

```diff
@@ -137,14 +137,9 @@ def _wrap_callback(callback, method=None):
     if callback is None or method in {'tnc', 'slsqp', 'cobyla'}:
         return callback  # don't wrap
 
-    p1 = (inspect.Parameter('intermediate_result',
-                            kind=inspect.Parameter.POSITIONAL_OR_KEYWORD),)
-    p2 = (inspect.Parameter('intermediate_result',
-                            kind=inspect.Parameter.KEYWORD_ONLY),)
-    sig1 = inspect.Signature(parameters=p1)
-    sig2 = inspect.Signature(parameters=p2)
-
-    if inspect.signature(callback) in {sig1, sig2}:
+    sig = inspect.signature(callback)
+
+    if set(sig.parameters) != {'intermediate_result'}:
```

Comment:
Oh wait it should be the other way around, my bad with my suggestion:
```suggestion
    if set(sig.parameters) == {'intermediate_result'}:
```